app/redux/store.js

```diff
@@ -34,6 +36,17 @@ const zmqMiddlewareCreator = (zmq, subzmq) => {
           zmq.connect('tcp://localhost:5556');
           subzmq.connect('tcp://localhost:5555');
 
+          dispatch({
+            type: 'CONNECTING'
+          });
+
+          connectTimer = setTimeout(() => {
+            dispatch({
+              type: 'ERROR',
+              message: 'Not able to connect to server'
+            });
+          }, 5000);
```

Comment:
Timeout `5000` could be moved to the config file (config file has been added in PR #11 I think)